### PR TITLE
Disable concurrent parts removal

### PR DESCRIPTION
**Pull request by alesapin**
### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):

### CODE REVIEW

**Review by tavplubix — APPROVED** [review]

File: src/Storages/MergeTree/MergeTreeData.cpp [hunk 1]
```diff
@@ -1916,7 +1916,8 @@ void MergeTreeData::clearPartsFromFilesystemImpl(const DataPartsVector & parts_t
     const auto settings = getSettings();
     if (parts_to_remove.size() > 1
         && settings->max_part_removal_threads > 1
-        && parts_to_remove.size() > settings->concurrent_part_removal_threshold)
+        && parts_to_remove.size() > settings->concurrent_part_removal_threshold
+        && (!supportsReplication() || !settings->allow_remote_fs_zero_copy_replication)) /// parts must be removed in order for zero-copy replication
```
Comment (alexey-milovidov) [inline src/Storages/MergeTree/MergeTreeData.cpp:1920]: The comment does not explain "why".

It reads like this:
```
/// parts must be removed in order for zero-copy replication
```
what the heck is "zero-copy" replication and why it requires part removal in order? Is it a kludge and is the code of "zero-copy" replication trashy?

**Review by alexey-milovidov — CHANGES_REQUESTED** [review]
.

**Comment by alexey-milovidov** [comment]
Related to https://github.com/ClickHouse/ClickHouse/pull/6438

**Comment by alexey-milovidov** [comment]
Why "not for changelog"?

**Review by alexey-milovidov — APPROVED** [review]

**Pull request merged**